include/libopencm3/sam/3n/pmc.h

```diff
@@ -0,0 +1,139 @@
+ *
+ * Copyright (C) 2015 Felix Held <felix-libopencm3@felixheld.de>
+ *
+ * This library is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this library.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef LIBOPENCM3_PMC_H
+#define LIBOPENCM3_PMC_H
+
+#include <libopencm3/sam/memorymap.h>
+#include <libopencm3/sam/common/pmc_common_all.h>
+#include <libopencm3/sam/common/pmc_common_3n3u.h>
+
+/* --- Power Management Controller (PMC) registers ----------------------- */
+
+/* PMC Clock Generator PLL Register */
+#define CKGR_PLLR			CKGR_PLLAR
+
+/* Oscillator Calibration Register */
+#define PMC_OCR				MMIO32(PMC_BASE + 0x0110)
+
+
+/* --- Register contents --------------------------------------------------- */
+
+
+/* --- PMC Clock Generator Main Oscillator Register (CKGR_MOR) ------------- */
+
+/* Wait Mode Command */
+#define CKGR_MOR_WAITMODE		(0x01 << 2)
+
+
+/* --- PMC Clock Generator PLL Register (CKGR_PLLR) ---------------------- */
+/* CKGR_PLLAR on all other device subfamilies */
+
+/* must be set to program CKGR_PLLR */
+#define CKGR_PLLR_ONE			CKGR_PLLAR_ONE
+
+/* PLLA Multiplier */
+#define CKGR_PLLR_MUL_SHIFT		CKGR_PLLAR_MULA_SHIFT
+#define CKGR_PLLR_MUL_MASK		CKGR_PLLAR_MULA_MASK
+
+/* PLLA Counter */
+#define CKGR_PLLR_PLLCOUNT_SHIFT	CKGR_PLLAR_PLLACOUNT_SHIFT
+#define CKGR_PLLR_PLLCOUNT_MASK		CKGR_PLLAR_PLLACOUNT_MASK
+
+/* Divider */
+#define CKGR_PLLR_DIV_SHIFT		CKGR_PLLAR_DIVA_SHIFT
+#define CKGR_PLLR_DIV_MASK		CKGR_PLLAR_DIVA_MASK
+
+
+/* --- PMC Master Clock Register (PMC_MCKR) -------------------------------- */
+
+/* PLL Divide by 2 */
+#define PMC_MCKR_PLLDIV2		(0x01 << 12)
+
+/* Master Clock Source Selection */
+#define PMC_MCKR_CSS_PLL_CLK		(2 << PMC_MCKR_CSS_SHIFT)
+
+
+/* --- PMC Programmable Clock Register 0 (PMC_PCK0) ------------------------ */
+
+/* Master Clock Source Selection */
+#define PMC_PCK0_CSS_PLL_CLK		(2 << PMC_PCK0_CSS_SHIFT)
+
+
+/* --- PMC Programmable Clock Register 1 (PMC_PCK1) ------------------------ */
+
+/* Master Clock Source Selection */
+#define PMC_PCK1_CSS_PLL_CLK		(2 << PMC_PCK1_CSS_SHIFT)
+
+
+/* --- PMC Programmable Clock Register 2 (PMC_PCK2) ------------------------ */
+
+/* Master Clock Source Selection */
+#define PMC_PCK2_CSS_PLL_CLK		(2 << PMC_PCK2_CSS_SHIFT)
+
+
+/* --- PMC Interrupt Enable Register (PMC_IER) ----------------------------- */
+
+/* PLL Lock Interrupt Enable */
+#define PMC_IER_LOCK			PMC_IER_LOCKA
```

Comment:
The PLLA is called PLL on the devices with only one PLL. I wanted to keep stuff consistent with the datasheet here.